pulsar-metadata/src/main/java/org/apache/pulsar/metadata/cache/impl/MetadataCacheImpl.java

```diff
@@ -58,6 +63,9 @@ public class MetadataCacheImpl<T> implements MetadataCache<T>, Consumer<Notifica
     private final MetadataStore store;
     private final MetadataStoreExtended storeExtended;
     private final MetadataSerde<T> serde;
+    private final ScheduledExecutorService backoffExecutor =
+            Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("metadata-cache-backoff"));
```

Comment:
Actually I'm wondering if we need one thread per `MetadataCache`. It means each time a `MetadataCache` instance is created, a new thread will be created, which might be unnecessary.

I think a better design is to add a common thread pool like `ForkJoinPool#commonPool` and a config in `MetadataCacheConfig` to specify a customized executor if users want.

Reply:
We can use the `AbstractMetadataStore.executor` as suggessted here: https://github.com/apache/pulsar/pull/23686#discussion_r1872973847